packages/angular/src/generators/library/lib/update-tsconfig.ts

```diff
@@ -39,7 +39,8 @@ function updateProjectIvyConfig(host: Tree, options: NormalizedSchema) {
       host,
       `${options.projectRoot}/tsconfig.lib.prod.json`,
       (json) => {
-        json.angularCompilerOptions['compilationMode'] = 'partial';
+        json.angularCompilerOptions['compilationMode'] =
+          options.compilationMode === 'full' ? undefined : 'partial';
```

Comment:
As per the docs: https://angular.io/guide/angular-compiler-options#compilationmode

Default is that if it is not set then it defaults to `full`. Would we not want to leave it unset (ie. using default) if we can, to reduce configuration lines that are unneeded?

Reply:
Setting it to `undefined` doesn't remove the property, isn't it? I agree on reducing config, but then we should instead delete the property if the `compilationMode` is `full`.

Reply:
Maybe setting it to undefined does remove the property and I'm wrong there, haven't tested it 🤔

Reply:
In the v13 migrations Nx sets the `enableIvy` value to undefined if it exists (which during migration removed it)

I am assuming that works because `undefined` is not a valid value for JSON, so when writing the JSON object in memory to a json file it is removed?

![image](https://user-images.githubusercontent.com/9469090/143487507-0f0f9712-cc03-49d4-bb9c-6a2afdd21ab5.png)

Reply:
Yep, you're right, it does get removed as it's not valid 🙂